### DISCUSSION TITLE
Changes to configurations for 8.0.0

### DISCUSSION THREAD

**Discussion by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#8914] [Community Feedback]
## Overview

We're working towards a new v8 major versions for typescript-eslint. Major versions are when we have an opportunity to adjust the [shared configs](https://typescript-eslint.io/users/configs). Exciting!

As with 6.0 (#6014), this is a proposal for the new shared configs. We're looking for feedback from the community before we go ahead and make the changes.

Please, let us know here or in the [typescript-eslint Discord's `#v8` channel](https://discord.com/channels/1026804805894672454/1228823100070821989): what feedback to you have? 💜

### Table Key

<table>
  <thead>
    <tr>
      <th>Column</th>
      <th>Description</th>
      <th>Emojis</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>Status</td>
      <td>Being added, deprecated, or removed</td>
      <td>
        <ul>
          <li>🆕 = newly added to typescript-eslint</li>
          <li>🙅 = to be deprecated in the next major</li>
          <li>💀 = currently deprecated; to be removed in the next version</li>
        </ul>
      </td>
    </tr>
    <tr>
      <td>TC</td>
      <td>Requires type checking?</td>
      <td>
        <ul>
          <li>💭 = yes</li>
        </ul>
      </td>
    </tr>
    <tr>
      <td>Ext</td>
      <td>Extension rule?</td>
      <td>
        <ul>
          <li>🧱 = yes</li>
        </ul>
      </td>
    </tr>
    <tr>
      <td>Rec'd</td>
      <td>Recommended</td>
      <td>
        <ul>
          <li>➕ = add to recommended this version</li>
          <li>➖️ = remove from recommended this version</li>
          <li>🟩 = stays in recommended this version</li>
        </ul>
      </td>
    </tr>
    <tr>
      <td>Strict</td>
      <td>Strict</td>
      <td>
        <ul>
        <li>➕ = add to strict this version</li>
        <li>➖️ = remove from strict this version</li>
        <li>🔵 = stays in strict this version</li>
        </ul>
      </td>
    </tr>
    <tr>
      <td>Style</td>
      <td>Style</td>
      <td>
        <ul>
          <li>➕ = add to stylistic this version</li>
          <li>➖️ = remove from stylistic this version</li>
          <li>🔸 = stays in stylistic this version</li>
        </ul>
      </td>
    </tr>
  </tbody>
</table>

### Recommendations Table

> Hint: search for 🆕 to find newly added rules, and ➕ or ➖ to see config changes.

| Rule                                                                                                                    | Status | TC  | Ext | Rec'd | Strict | Style | Comment                                                                                                                       |
| ----------------------------------------------------------------------------------------------------------------------- | ------ | --- | --- | ----- | ------ | ----- | ----------------------------------------------------------------------------------------------------------------------------- |
| [`adjacent-overload-signatures`](https://typescript-eslint.io/rules/adjacent-overload-signatures)                       |        |     |     |       |        | 🔸    |
| [`array-type`](https://typescript-eslint.io/rules/array-type)                                                           |        |     |     |       |        | 🔸    |
| [`await-thenable`](https://typescript-eslint.io/rules/await-thenable)                                                   |        | 💭  |     | 🟩    |        |       |
| [`ban-ts-comment`](https://typescript-eslint.io/rules/ban-ts-comment)                                                   |        |     |     | 🟩    |        |       |
| [`ban-tslint-comment`](https://typescript-eslint.io/rules/ban-tslint-comment)                                           |        |     |     |       |        | 🔸    |
| [`ban-types`](https://typescript-eslint.io/rules/ban-types)                                                             | 🙅     |     |     | ➖️   |        |       | See [Replacement of `ban-types`](https://typescript-eslint.io/blog/announcing-typescript-eslint-v8/#replacement-of-ban-types) |
| [`no-restricted-types`](https://typescript-eslint.io/rules/no-restricted-types)                                         |        |     |     | ➕    |        |       |
| [`no-unsafe-function-type`](https://typescript-eslint.io/rules/no-unsafe-function-type)                                 |        |     |     | ➕    |        |       |
| [`no-wrapper-object-types`](https://typescript-eslint.io/rules/no-wrapper-object-types)                                 |        |     |     | ➕    |        |       |
| [`class-literal-property-style`](https://typescript-eslint.io/rules/class-literal-property-style)                       |        |     |     |       |        | 🔸    |
| [`class-methods-use-this`](https://typescript-eslint.io/rules/class-methods-use-this)                                   | 🆕     |     | 🧱  |       |        |       | too opinionated for stylistic                                                                                                 |
| [`consistent-generic-constructors`](https://typescript-eslint.io/rules/consistent-generic-constructors)                 |        |     |     |       |        | 🔸    |
| [`consistent-indexed-object-style`](https://typescript-eslint.io/rules/consistent-indexed-object-style)                 |        |     |     |       |        | 🔸    |
| [`consistent-return`](https://typescript-eslint.io/rules/consistent-return)                                             | 🆕     | 💭  | 🧱  |       |        |       | too opinionated for stylistic                                                                                                 |
| [`consistent-type-assertions`](https://typescript-eslint.io/rules/consistent-type-assertions)                           |        |     |     |       |        | 🔸    |
| [`consistent-type-definitions`](https://typescript-eslint.io/rules/consistent-type-definitions)                         |        |     |     |       |        | 🔸    |
| [`consistent-type-exports`](https://typescript-eslint.io/rules/consistent-type-exports)                                 |        | 💭  |     |       |        |       |
| [`consistent-type-imports`](https://typescript-eslint.io/rules/consistent-type-imports)                                 |        |     |     |       |        |       |
| [`default-param-last`](https://typescript-eslint.io/rules/default-param-last)                                           |        |     | 🧱  |       |        |       |
| [`dot-notation`](https://typescript-eslint.io/rules/dot-notation)                                                       |        | 💭  | 🧱  |       |        | 🔸    |
| [`explicit-function-return-type`](https://typescript-eslint.io/rules/explicit-function-return-type)                     |        |     |     |       |        |       |
| [`explicit-member-accessibility`](https://typescript-eslint.io/rules/explicit-member-accessibility)                     |        |     |     |       |        |       |
| [`explicit-module-boundary-types`](https://typescript-eslint.io/rules/explicit-module-boundary-types)                   |        |     |     |       |        |       |
| [`init-declarations`](https://typescript-eslint.io/rules/init-declarations)                                             |        |     | 🧱  |       |        |       |
| [`max-params`](https://typescript-eslint.io/rules/max-params)                                                           | 🆕     |     | 🧱  |       |        |       | too opinionated for stylistic                                                                                                 |
| [`member-ordering`](https://typescript-eslint.io/rules/member-ordering)                                                 |        |     |     |       |        |       |
| [`method-signature-style`](https://typescript-eslint.io/rules/method-signature-style)                                   |        |     |     |       |        |       |
| [`naming-convention`](https://typescript-eslint.io/rules/naming-convention)                                             |        | 💭  |     |       |        |       |
| [`no-array-constructor`](https://typescript-eslint.io/rules/no-array-constructor)                                       |        |     | 🧱  | 🟩    |        |       |
| [`no-array-delete`](https://typescript-eslint.io/rules/no-array-delete)                                                 | 🆕     | 💭  |     | ➕    | ➖     |       |
| [`no-base-to-string`](https://typescript-eslint.io/rules/no-base-to-string)                                             |        | 💭  |     | 🟩    |        |       |
| [`no-confusing-non-null-assertion`](https://typescript-eslint.io/rules/no-confusing-non-null-assertion)                 |        |     |     |       |        | 🔸    |
| [`no-confusing-void-expression`](https://typescript-eslint.io/rules/no-confusing-void-expression)                       |        | 💭  |     |       |        |       |
| [`no-dupe-class-members`](https://typescript-eslint.io/rules/no-dupe-class-members)                                     |        |     | 🧱  |       |        |       |
| [`no-duplicate-enum-values`](https://typescript-eslint.io/rules/no-duplicate-enum-values)                               |        |     |     | 🟩    |        |       |
| [`no-duplicate-type-constituents`](https://typescript-eslint.io/rules/no-duplicate-type-constituents)                   |        | 💭  |     | 🟩    |        |       |
| [`no-dynamic-delete`](https://typescript-eslint.io/rules/no-dynamic-delete)                                             |        |     |     |       | 🔵     |       |
| [`no-empty-function`](https://typescript-eslint.io/rules/no-empty-function)                                             |        |     | 🧱  |       |        | 🔸    |
| [`no-empty-object-type`](https://v8--typescript-eslint.netlify.app/rules/no-empty-object-type)                          | 🆕     |     |     | ➕    |        |       |
| [`no-empty-interface`](https://typescript-eslint.io/rules/no-empty-interface)                                           | 🙅     |     |     |       |        | ➖    |
| [`no-explicit-any`](https://typescript-eslint.io/rules/no-explicit-any)                                                 |        |     |     | 🟩    |        |       |
| [`no-extra-non-null-assertion`](https://typescript-eslint.io/rules/no-extra-non-null-assertion)                         |        |     |     | 🟩    |        |       |
| [`no-extra-semi`](https://typescript-eslint.io/rules/no-extra-semi)                                                     |        |     | 🧱  |       |        |       |
| [`no-extraneous-class`](https://typescript-eslint.io/rules/no-extraneous-class)                                         |        |     |     |       | 🔵     |       |
| [`no-floating-promises`](https://typescript-eslint.io/rules/no-floating-promises)                                       |        | 💭  |     | 🟩    |        |       |
| [`no-for-in-array`](https://typescript-eslint.io/rules/no-for-in-array)                                                 |        | 💭  |     | 🟩    |        |       |
| [`no-implied-eval`](https://typescript-eslint.io/rules/no-implied-eval)                                                 |        | 💭  | 🧱  | 🟩    |        |       |
| [`no-import-type-side-effects`](https://typescript-eslint.io/rules/no-import-type-side-effects)                         |        |     |     |       |        |       |
| [`no-inferrable-types`](https://typescript-eslint.io/rules/no-inferrable-types)                                         |        |     |     |       |        | 🔸    |
| [`no-invalid-this`](https://typescript-eslint.io/rules/no-invalid-this)                                                 |        |     | 🧱  |       |        |       |
| [`no-invalid-void-type`](https://typescript-eslint.io/rules/no-invalid-void-type)                                       |        |     |     |       | 🔵     |       |
| [`no-loop-func`](https://typescript-eslint.io/rules/no-loop-func)                                                       |        |     | 🧱  |       |        |       |
| [`no-loss-of-precision`](https://typescript-eslint.io/rules/no-loss-of-precision)                                       | 🙅     |     | 🧱  | ➖    |        |       | No longer necessary                                                                                                           |
| [`no-magic-numbers`](https://typescript-eslint.io/rules/no-magic-numbers)                                               |        |     | 🧱  |       |        |       |
| [`no-meaningless-void-operator`](https://typescript-eslint.io/rules/no-meaningless-void-operator)                       |        | 💭  |     |       | 🔵     |       |
| [`no-misused-new`](https://typescript-eslint.io/rules/no-misused-new)                                                   |        |     |     | 🟩    |        |       |
| [`no-misused-promises`](https://typescript-eslint.io/rules/no-misused-promises)                                         |        | 💭  |     | 🟩    |        |       |
| [`no-mixed-enums`](https://typescript-eslint.io/rules/no-mixed-enums)                                                   |        | 💭  |     |       | 🔵     |       |
| [`no-namespace`](https://typescript-eslint.io/rules/no-namespace)                                                       |        |     |     | 🟩    |        |       |
| [`no-non-null-asserted-nullish-coalescing`](https://typescript-eslint.io/rules/no-non-null-asserted-nullish-coalescing) |        |     |     |       | 🔵     |       |
| [`no-non-null-asserted-optional-chain`](https://typescript-eslint.io/rules/no-non-null-asserted-optional-chain)         |        |     |     | 🟩    |        |       |
| [`no-non-null-assertion`](https://typescript-eslint.io/rules/no-non-null-assertion)                                     |        |     |     |       | 🔵     |       |
| [`no-redeclare`](https://typescript-eslint.io/rules/no-redeclare)                                                       |        |     | 🧱  |       |        |       |
| [`no-redundant-type-constituents`](https://typescript-eslint.io/rules/no-redundant-type-constituents)                   |        | 💭  |     | 🟩    |        |       |
| [`no-require-imports`](https://typescript-eslint.io/rules/no-require-imports)                                           |        |     |     |       |        |       |
| [`no-restricted-imports`](https://typescript-eslint.io/rules/no-restricted-imports)                                     |        |     | 🧱  |       |        |       |
| [`no-shadow`](https://typescript-eslint.io/rules/no-shadow)                                                             |        |     | 🧱  |       |        |       |
| [`no-this-alias`](https://typescript-eslint.io/rules/no-this-alias)                                                     |        |     |     | 🟩    |        |       |
| [`no-throw-literal`](https://typescript-eslint.io/rules/no-throw-literal)                                               |        | 💭  | 🧱  |       |        |       |
| [`no-type-alias`](https://typescript-eslint.io/rules/no-type-alias)                                                     |        |     |     |       |        |       |
| [`no-unnecessary-boolean-literal-compare`](https://typescript-eslint.io/rules/no-unnecessary-boolean-literal-compare)   |        | 💭  |     |       | 🔵     |       |
| [`no-unnecessary-condition`](https://typescript-eslint.io/rules/no-unnecessary-condition)                               |        | 💭  |     |       | 🔵     |       |
| [`no-unnecessary-qualifier`](https://typescript-eslint.io/rules/no-unnecessary-qualifier)                               |        | 💭  |     |       |        |       |
| [`no-unnecessary-type-arguments`](https://typescript-eslint.io/rules/no-unnecessary-type-arguments)                     |        | 💭  |     |       | 🔵     |       |
| [`no-unnecessary-type-assertion`](https://typescript-eslint.io/rules/no-unnecessary-type-assertion)                     |        | 💭  |     | 🟩    |        |       |
| [`no-unnecessary-type-constraint`](https://typescript-eslint.io/rules/no-unnecessary-type-constraint)                   |        |     |     | 🟩    |        |       |
| [`no-unsafe-argument`](https://typescript-eslint.io/rules/no-unsafe-argument)                                           |        | 💭  |     | 🟩    |        |       |
| [`no-unsafe-assignment`](https://typescript-eslint.io/rules/no-unsafe-assignment)                                       |        | 💭  |     | 🟩    |        |       |
| [`no-unsafe-call`](https://typescript-eslint.io/rules/no-unsafe-call)                                                   |        | 💭  |     | 🟩    |        |       |
| [`no-unsafe-declaration-merging`](https://typescript-eslint.io/rules/no-unsafe-declaration-merging)                     |        |     |     | 🟩    |        |       |
| [`no-unsafe-enum-comparison`](https://typescript-eslint.io/rules/no-unsafe-enum-comparison)                             |        | 💭  |     | 🟩    |        |       |
| [`no-unsafe-member-access`](https://typescript-eslint.io/rules/no-unsafe-member-access)                                 |        | 💭  |     | 🟩    |        |       |
| [`no-unsafe-return`](https://typescript-eslint.io/rules/no-unsafe-return)                                               |        | 💭  |     | 🟩    |        |       |
| [`no-unsafe-unary-minus`](https://typescript-eslint.io/rules/no-unsafe-unary-minus)                                     | 🆕     | 💭  |     | ➕    |        |       |
| [`no-unused-expressions`](https://typescript-eslint.io/rules/no-unused-expressions)                                     |        |     | 🧱  | ➕    |        |       |
| [`no-unused-vars`](https://typescript-eslint.io/rules/no-unused-vars)                                                   |        |     | 🧱  | 🟩    |        |       |
| [`no-use-before-define`](https://typescript-eslint.io/rules/no-use-before-define)                                       |        |     | 🧱  |       |        |       |
| [`no-useless-constructor`](https://typescript-eslint.io/rules/no-useless-constructor)                                   |        |     | 🧱  |       | 🔵     |       |
| [`no-useless-empty-export`](https://typescript-eslint.io/rules/no-useless-empty-export)                                 |        |     |     |       |        |       |
| [`no-useless-template-literals`](https://typescript-eslint.io/rules/no-useless-template-literals)                       | 🆕     | 💭  |     | ➕    | ➖     |       | To be renamed in #8821                                                                                                        |
| [`no-var-requires`](https://typescript-eslint.io/rules/no-var-requires)                                                 | 🙅     |     |     | ➖    |        |       | Switched to `no-require-imports`                                                                                              |
| [`non-nullable-type-assertion-style`](https://typescript-eslint.io/rules/non-nullable-type-assertion-style)             |        | 💭  |     |       |        | 🔸    |
| [`only-throw-error`](https://typescript-eslint.io/rules/only-throw-error)                                               | 🆕     | 💭  | 🧱  | ➕    | ➖     |       |
| [`parameter-properties`](https://typescript-eslint.io/rules/parameter-properties)                                       |        |     |     |       |        |       | too opinionated for stylistic                                                                                                 |
| [`prefer-as-const`](https://typescript-eslint.io/rules/prefer-as-const)                                                 |        |     |     | 🟩    |        |       |
| [`prefer-destructuring`](https://typescript-eslint.io/rules/prefer-destructuring)                                       | 🆕     | 💭  | 🧱  |       |        |       | too opinionated for stylistic                                                                                                 |
| [`prefer-enum-initializers`](https://typescript-eslint.io/rules/prefer-enum-initializers)                               |        |     |     |       |        |       |
| [`prefer-find`](https://typescript-eslint.io/rules/prefer-find)                                                         | 🆕     | 💭  |     |       |        | ➕    |
| [`prefer-for-of`](https://typescript-eslint.io/rules/prefer-for-of)                                                     |        |     |     |       |        | 🔸    |
| [`prefer-function-type`](https://typescript-eslint.io/rules/prefer-function-type)                                       |        |     |     |       |        | 🔸    |
| [`prefer-includes`](https://typescript-eslint.io/rules/prefer-includes)                                                 |        | 💭  |     |       | 🔵     | ➕    |
| [`prefer-literal-enum-member`](https://typescript-eslint.io/rules/prefer-literal-enum-member)                           |        |     |     |       | 🔵     |       |
| [`prefer-namespace-keyword`](https://typescript-eslint.io/rules/prefer-namespace-keyword)                               |        |     |     | ➕    |        | ➖    |
| [`prefer-nullish-coalescing`](https://typescript-eslint.io/rules/prefer-nullish-coalescing)                             |        | 💭  |     |       |        | 🔸    |
| [`prefer-optional-chain`](https://typescript-eslint.io/rules/prefer-optional-chain)                                     |        | 💭  |     |       |        | 🔸    |
| [`prefer-promise-reject-errors`](https://typescript-eslint.io/rules/prefer-promise-reject-errors)                       | 🆕     | 💭  | 🧱  | ➕    | ➖     |       |
| [`prefer-readonly`](https://typescript-eslint.io/rules/prefer-readonly)                                                 |        | 💭  |     |       |        |       |
| [`prefer-readonly-parameter-types`](https://typescript-eslint.io/rules/prefer-readonly-parameter-types)                 |        | 💭  |     |       |        |       |
| [`prefer-reduce-type-parameter`](https://typescript-eslint.io/rules/prefer-reduce-type-parameter)                       |        | 💭  |     |       | 🔵     |       |
| [`prefer-regexp-exec`](https://typescript-eslint.io/rules/prefer-regexp-exec)                                           |        | 💭  |     |       |        | ➕    |
| [`prefer-return-this-type`](https://typescript-eslint.io/rules/prefer-return-this-type)                                 |        | 💭  |     |       | 🔵     |       |
| [`prefer-string-starts-ends-with`](https://typescript-eslint.io/rules/prefer-string-starts-ends-with)                   |        | 💭  |     |       |        | 🔸    |
| [`prefer-ts-expect-error`](https://typescript-eslint.io/rules/prefer-ts-expect-error)                                   |        |     |     |       | 🔵     |       |
| [`promise-function-async`](https://typescript-eslint.io/rules/promise-function-async)                                   |        | 💭  |     |       |        |       |
| [`require-array-sort-compare`](https://typescript-eslint.io/rules/require-array-sort-compare)                           |        | 💭  |     |       |        |       |
| [`require-await`](https://typescript-eslint.io/rules/require-await)                                                     |        | 💭  | 🧱  | 🟩    |        |       |
| [`restrict-plus-operands`](https://typescript-eslint.io/rules/restrict-plus-operands)                                   |        | 💭  |     | 🟩    |        |       |
| [`restrict-template-expressions`](https://typescript-eslint.io/rules/restrict-template-expressions)                     |        | 💭  |     | 🟩    |        |       |
| [`return-await`](https://typescript-eslint.io/rules/return-await)                                                       |        | 💭  | 🧱  |       | ➕     |       |
| [`sort-type-constituents`](https://typescript-eslint.io/rules/sort-type-constituents)                                   |        |     |     |       |        |       |
| [`strict-boolean-expressions`](https://typescript-eslint.io/rules/strict-boolean-expressions)                           |        | 💭  |     |       |        |       |
| [`switch-exhaustiveness-check`](https://typescript-eslint.io/rules/switch-exhaustiveness-check)                         |        | 💭  |     |       |        |       |
| [`triple-slash-reference`](https://typescript-eslint.io/rules/triple-slash-reference)                                   |        |     |     | 🟩    |        |       |
| [`typedef`](https://typescript-eslint.io/rules/typedef)                                                                 |        |     |     |       |        |       |
| [`unbound-method`](https://typescript-eslint.io/rules/unbound-method)                                                   |        | 💭  |     | 🟩    |        |       |
| [`unified-signatures`](https://typescript-eslint.io/rules/unified-signatures)                                           |        |     |     |       | 🔵     |       |
| [`use-unknown-in-catch-callback-variable`](https://typescript-eslint.io/rules/use-unknown-in-catch-callback-variable)   | 🆕     | 💭  |     |       | 🔵     |       | too opinionated for recommended                                                                                               |

**Comment by JoshuaKGoldberg** [comment]
Fun fact, I somewhat auto-generated this table. 😎 https://github.com/JoshuaKGoldberg/repros/tree/ts-eslint-v8-config-changes-generator

**Comment by kirkwaiblinger** [comment]
If no-useless-template-literal is destined to be renamed, should we sneak that in before it gets added to recommended to minimize impact of the renaming (on things like disable comments etc)?


https://github.com/typescript-eslint/typescript-eslint/pull/8821

**Reply by JoshuaKGoldberg** [reply]
Agreed, added to the v8 milestone. Good call!

**Comment by JoshuaKGoldberg** [comment]
Interesting note: [prefer-find](https://typescript-eslint.io/rules/prefer-find) and [prefer-includes](https://typescript-eslint.io/rules/prefer-includes) at first seem like rules that would be in `stylistic*`. `prefer-includes` is currently in `strict*`. I don't mind keeping them in `strict` because they do help code be a little more performant... but I suspect the main benefits for users are stylistic. It'd be very rare for actual perf benefits to come out of them. So I'm leaning towards moving them to `stylistic*`. Does anybody have thoughts on this?

**Reply by kirkwaiblinger** [reply]
The way I see it, both those rules satisfy the conditions when flagging that "there is a unique and uncontroversially better way to write this, though there is not a significant correctness concern with the existing code". So there's no reason ever to not want them, but they're not terribly important to use either. As a user my intuition would that that should be part of strict, whereas stylistic would have to do more with opinionated things that don't have a uniquely better way to write them; one is just picked largely for the sake of consistency (consistent-type-assertions springs to mind as a good example of this). Idk if that's really the intention of those configs or not, though.

**Reply by rubiesonthesky** [reply]
I think it would be good to also consider that strict is recommended only for proficient typescript developers. And I think these rules are easy to use and easy to adopt. So they could go for stylistics in that sense. 

I think `prefer-find` could go to even recommended. I don't see it being really opinionated. Instead it will catch old code that could be updated. There is probably only handful of cases where you would not want to use that rule.

> We recommend a TypeScript project extend from plugin:@typescript-eslint/strict only if a nontrivial percentage of its developers are highly proficient in TypeScript.

**Reply by JoshuaKGoldberg** [reply]
I'm leaning towards `stylistic` for now, as (a) they really do mostly give objectively better stylistic benefits and (b) per @rubiesonthesky's note, `strict` locks them behind _"you should be proficient"_ notes.

**Comment by hyoban** [comment]
Here are some rules I find handy

Add `no-import-type-side-effects` to the recommended config, and `consistent-type-exports` with `consistent-type-imports` to the stylistic config.

- https://typescript-eslint.io/blog/consistent-type-imports-and-exports-why-and-how
- https://youtu.be/5Uc6qDR10Vk?si=9RHfxoAVd2mxcmQv

`method-signature-style`

- https://www.totaltypescript.com/method-shorthand-syntax-considered-harmful

**Reply by bradzacher** [reply]
`consistent-type-imports` might too opinionated for us to turn on, I think.
Mainly because it has a fixer and it fixes to an opinionated default.

`stylistic` is our opinionated default but I am weary of complaints.
We could document that people should use [`import/consistent-type-specifier-style`](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/consistent-type-specifier-style.md) if they want to customise the style...

🤔

**Reply by bradzacher** [reply]
`no-import-type-side-effects` isn't really useful for people unless they are using `verbatimModuleSyntax` as well. With that turned off the "empty imports" are all properly elided by everything.

It's a useful rule if you have to use `verbatimModuleSyntax` - otherwise it's just noise and not necessary.

**Comment by rubiesonthesky** [comment]
Just heads up, ➕ and ➖ emojis are not visible on dark mode.

**Reply by JoshuaKGoldberg** [reply]
Oh, blurgh. Good callout. Now we have to figure out - what's a good replacement pair of emojis... 🤔

**Reply by rubiesonthesky** [reply]
remove: 🗑️ (or ❌ but that is too red IMO.)
add: 🎉 / ✨

(I honestly do not know, I think it could be just `+` and `-` :)

**Reply by JoshuaKGoldberg** [reply]
Mmhh it'd be nice to have two emojis with meanings related to each other... surprisingly hard to find. I'm surprised there aren't + and - versions in squares the way there are 🔼 / 🔽 .

**Reply by rubiesonthesky** [reply]
👍 /  👎 :D  There is ✅ and ❎ but I think those are hard for some people as they use same color and the different symbol is just so small. 

I think the table could use ✨ for meaning "new". Both meaning as a new rule and as a new item in shared config. Because I had difficulties remembering what every emoji means :D But I'll leave this to you (or others) to find best option. 

(Found this kind of tool where you could check in both modes easily https://github-emoji-picker.rickstaa.dev/ but emojis are pretty limited in this way)

**Comment by rubiesonthesky** [comment]
### `prefer-namespace-keyword`
`module` will be probably marked as deprecated soon in typescript https://github.com/microsoft/TypeScript/issues/57913 

I wonder would it be helpful for ts-eslint users that this rule would be moved to recommended so they would get heads up beforehand? Of course, marking it as deprecated does not mean that it will stop working. :)

**Reply by JoshuaKGoldberg** [reply]
I like it! It straddles the line between "recommended" and "stylistic"... but given the strong TypeScript push, that's enough for me to be happy switching it over.

**Comment by JoshuaKGoldberg** [comment]
Update (May 11th): with #8977's `no-empty-object-type` merged into `v8`, updating the table to include that being added to `recommended`.

**Reply by JoshuaKGoldberg** [reply]
Note that once #9102 is merged, we'll have some more changes to the table.

**Comment by caribou-code** [comment]
I'm excited to see all this work on v8! Is there a rough timeline for when this version will drop? I think I'm right in saying ESLint v9 has been out nearly a couple of months now but the current version of typescript-eslint doesn't support it, preventing new users/projects from using the latest major version.

**Reply by bradzacher** [reply]
No strict timeline. When we get to things, really.
You can use the canary releases if you need v9 support now.

As volunteer maintainers we work on things when we have the bandwidth -- and unfortunately we haven't had the bandwidth. ESLint sadly did not coordinate with us and released before we were ready and at a time where we have been time poor.

**Comment by JoshuaKGoldberg** [comment]
Coming over from https://github.com/typescript-eslint/typescript-eslint/pull/9079#discussion_r1615395842: we're going to try out moving [@typescript-eslint/no-unused-expressions](https://typescript-eslint.io/rules/no-unused-expressions) to `recommended`. It's analogous to [@typescript-eslint/no-unused-vars](https://typescript-eslint.io/rules/no-unused-vars).

**Comment by kirkwaiblinger** [comment]
Calling out that return-await is on track to be added to the strict config, a la #8667/#9604

**Comment by rubiesonthesky** [comment]
I'm coming from blog post https://typescript-eslint.io/blog/announcing-typescript-eslint-v8 to see what where the actual changes to some rules. :)

Some updates that could be made to table or just noting these here for next user that is curious why they are listed removed in config diffs.  Though if you read the blog lost further, you also get the same explanation :D 

- `ban-types` was deprecated 
- `no-loss-of-precision` was deprecated
- `no-var-requires` was deprecated

**Reply by JoshuaKGoldberg** [reply]
Aha thank you, great callout! Updating now.
